api/app/Models/PoolCandidate.php

```diff
@@ -1066,62 +1070,79 @@ public function computeAssessmentStatus()
                 continue;
             }
 
-            if ($hasEssentialSkillsToAssess) {
+            foreach ($step->poolSkills as $poolSkill) {
+                $result = $stepResults->firstWhere('pool_skill_id', $poolSkill->id);
+                $decision = $result?->assessment_decision;
 
-                // Check assessed essential skills on this step
-                $essentialSkillAssessments = $stepResults
-                    ->filter(function ($result) {
-                        return $result->poolSkill?->type === PoolSkillType::ESSENTIAL->name &&
-                            $result->assessment_result_type !== AssessmentResultType::EDUCATION;
-                    });
+                if ($poolSkill->type === PoolSkillType::ESSENTIAL->name) {
+                    if (! $result || is_null($result->assessment_decision)) {
+                        $hasToAssess = true;
 
-                if ($essentialSkillAssessments->isEmpty()) {
-                    $hasToAssess = true;
-                }
+                        continue;
+                    }
 
-                foreach ($essentialSkillAssessments as $essentialSkillAssessment) {
-                    $decision = $essentialSkillAssessment->assessment_decision;
+                    if ($decision === AssessmentDecision::UNSUCCESSFUL->name) {
+                        $hasFailure = true;
+                        break;
+                    }
+
+                    if ($decision === AssessmentDecision::HOLD->name) {
+                        $hasOnHold = true;
 
-                    switch ($decision) {
-                        case null:
+                        continue;
+                    }
+                } else {
+                    if ($poolSkill->skill->category === SkillCategory::TECHNICAL->name) {
+                        $isClaimed = false;
+                        $snapshot = $this->profile_snapshot;
+
+                        if ($snapshot) {
+                            $claimedSkills = collect($snapshot['userSkills']);
+                            $isClaimed = $claimedSkills->contains(function ($userSkill) use ($poolSkill) {
+                                return $userSkill['skill']['id'] === $poolSkill->skill_id;
+                            });
+                        }
+
+                        if (! $isClaimed) {
+                            continue;
+                        }
+
+                        if (! $result || is_null($result->assessment_decision)) {
                             $hasToAssess = true;
-                            break;
-                        case AssessmentDecision::HOLD->name:
-                            $hasOnHold = true;
-                            break;
-                        case AssessmentDecision::UNSUCCESSFUL->name:
-                            $hasFailure = true;
-                            break;
-                        default:
+
+                            continue;
+                        }
```

Comment:
This should be outside the TECHNICAL if branch, since it needs to apply to asset behvioural skills.

Reply:
Oops, yeah good catch! Moved in [remove some guards, fix conditions, add tests](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10925/commits/fb446bf12a5a1196542241022b8aa873527da26b)